django/db/migrations/state.py

```diff
@@ -680,10 +680,18 @@ def clone(self):
         """Return a clone of this registry."""
         clone = StateApps([], {})
         clone.all_models = copy.deepcopy(self.all_models)
-        clone.app_configs = copy.deepcopy(self.app_configs)
-        # Set the pointer to the correct app registry.
-        for app_config in clone.app_configs.values():
-            app_config.apps = clone
+
+        clone.app_configs = {}
+        for app_label, app_config in self.app_configs.items():
+            current_apps = app_config.apps
+            current_models = app_config.models
+            app_config.apps, app_config.models = None, None
+            clone.app_configs[app_label] = copy.deepcopy(app_config)
```

Comment:
Another way to do it is implementing a `AppConfigStub.__deepcopy__` with this logic.